### ISSUE TITLE
Allow workbench.action.tasks.runTask as input command variable

### ISSUE THREAD

**Issue by vishalwy** [microsoft/vscode#73877]
I have two shell tasks defined in my tasks.json; **build** and **get-exe-path**. Both the tasks gets the ${input:config} and ${input:target} variables from the user and pass on as arguments to the shell command. **build** task executes CMake & make to build the target. **get-exe-path** finds the executable target generated and prints out the path. 

Now in my launch.json, I have set "preLaunchTask" to **build**, and it runs properly. But I also want "program" field to execute **get-exe-path** to get the actual exe path. For this I have used **workbench.action.tasks.runTask** as input command variable. But it is not working as expected. It does start the task **get-exe-path**, but the debugging ends right there. Seems like **workbench.action.tasks.runTask** is not returning a thenable with the command output.

**Comment by alexr00** [comment]
I like the idea here, but tasks don't have a real return value. They can run scripts. Then can run code. There can be any amount of stuff output to the terminal, so it's not clear what should be returned as the "result" of task. The result of a task is an exit code, and we don't surface that as a result of `workbench.action.tasks.runTask`. 
Instead, you could write an extension that provides the result value. Or, you could set the path you need in an environment variable and reference the environment variable in your launch.json.

**Comment by vishalwy** [comment]
Well, I would expect the input command variable to work the same way Bash's command substitution works.

Anyway, I ended up writing an extension. https://github.com/vishalwy/vsc-tasker. And I ran into couple of issues while developing it.

Here is how the extension captures the output from the task. 

1. Give the task a unique name, for eg, tasker@timestamp
2. Attach `vscode.tasks.onDidStartTask`, `vscode.tasks.onDidEndTask` and start the task.
3. On `vscode.tasks.onDidStartTask`, identify the terminal for the task.
  3.1. Loop through `vscode.window.terminals` to find out the terminal matching the task name.
  3.2.  If no matching terminal found, attach `vscode.window.onDidOpenTerminal` to match against new terminals getting created.
4. Once a matching terminal is found, attach `terminal.onDidWriteData` to capture the output.
5. On `terminal.onDidWriteData`, store the output.
6. On `vscode.tasks.onDidEndTask` resolve the promise with the output.

https://github.com/vishalwy/vsc-tasker/blob/master/src/run.ts#L90

**Issue 1:**
If 3.2 happens, then step 5 won't execute. 

The workaround for the time being is to run a pseudo task before the actual task to force create a terminal. And then set the task's `presentationOptions.panel` to `vscode.TaskPanelKind.Shared` so that it uses the existing terminal.

**Issue 2:**
If the task has input type `promptString` or `pickString`, and if the user cancels the input, neither `vscode.tasks.onDidStartTask` nor `vscode.tasks.onDidEndTask` fires leaving the extension command in an unresolved state. 

**VS Code 1.35.0 (1.35.0)**
**MacOS 10.14.2**
